### PR TITLE
Adding sts support

### PR DESCRIPTION
**Pull request by maorlx**

### CODE REVIEW

File: epsagon/aws_sdk_v2_factories/sts_factory.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+func updateMetadataField(data reflect.Value, key string, res *protocol.Resource) {
+	value, ok := getFieldStringPtr(data, key)
+	if ok {
+		res.Metadata[key] = value
+	}
+}
+
+func handleStsGetCallerIdentityRequest(r *aws.Request, res *protocol.Resource, metadataOnly bool) {
+	outputValue := reflect.ValueOf(r.Data).Elem()
```
Comment (Alon-Katz) [inline epsagon/aws_sdk_v2_factories/sts_factory.go:25]: Do we need this if metadataOnly is true?

File: epsagon/aws_sdk_v2_factories/sts_factory.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+// STSEventDataFactory to create epsagon Resource from aws.Request to STS
+func StsDataFactory(r *aws.Request, res *protocol.Resource, metadataOnly bool) {
+	handleSpecificOperations := map[string]specificOperationHandler{
+		"GetCallerIdentity": handleStsGetCallerIdentityRequest,
+	}
+	handleSpecificOperation(r, res, metadataOnly, handleSpecificOperations, nil)
+}
+
+func updateMetadataField(data reflect.Value, key string, res *protocol.Resource) {
```
Comment (enoodle) [inline epsagon/aws_sdk_v2_factories/sts_factory.go:17]: put this function in aws_sdk_v2_factories/common_utils.go

**Comment by adavidai** [comment]
Please add a usage example

File: epsagon/aws_sdk_v2_factories/sts_factory.go (hunk 1 above)
Comment (maorlx) [reply epsagon/aws_sdk_v2_factories/sts_factory.go:18]: yes as this data shouldn't be sent if metadataOnly is true

Comment (adavidai) [reply epsagon/aws_sdk_v2_factories/sts_factory.go:18]: I think what @Alon-Katz meant is that you can add this line into the if !metadataOnly (as you do not need to execute it if metadataOnly is true)

Comment (maorlx) [reply epsagon/aws_sdk_v2_factories/sts_factory.go:18]: oh now I see. Right, will fix

**Pull request merged**